### ISSUE TITLE
Potentially incorrect 'negative' sign in regularization_function.py

### ISSUE THREAD

**Issue by pranavm1502** [SchusterLab/quantum-optimal-control#10]
Hi,

In regularization_function.py, I think the following negative sign needs to be corrected. `speed_up_reg_alpha_coeff = - tfs.sys_para.reg_coeffs['speed_up']` to `speed_up_reg_alpha_coeff = tfs.sys_para.reg_coeffs['speed_up']`. Having the negative sign, makes the loss function negative and leads to error in convergence.

**Comment by nelsonleung** [comment]
Hi,

We can change it to positive and modify the last line to be
`reg_loss = reg_loss + speed_up_reg_alpha * tf.nn.l2_loss(tf.ones(tfs.sys_para.steps+1)-target_vecs_inner_product)`. This will preserve the desired functionality (fig 5b of the paper) without having the convergence issue. Thanks for pointing it out. I will soon make an update on this.

**Comment by nelsonleung** [comment]
fixed 7cbf21896bd4edc618b1470b708ce5b3d56579d2
